cumulus/polkadot-parachain/src/common/solochain.rs

```diff
@@ -0,0 +1,263 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::common::{
+	BuildImportQueue, BuildSelectChain, ConstructNodeRuntimeApi, NodeBackend, NodeSpec,
+	NodeSpecProvider, RpcModule, ServiceError, ServiceResult,
+};
+use futures::FutureExt;
+use sc_client_api::Backend;
+use sc_consensus::{DefaultImportQueue, LongestChain};
+use sc_executor::WasmExecutor;
+use sc_network::NetworkBackend;
+use sc_service::{Configuration, PartialComponents, TFullClient, TaskManager};
+use sc_telemetry::TelemetryHandle;
+use sc_transaction_pool_api::OffchainTransactionPoolFactory;
+use sp_core::crypto::AccountId32;
+use sp_runtime::{generic, traits, traits::Block as BlockT, OpaqueExtrinsic};
+use std::{marker::PhantomData, sync::Arc};
+use substrate_frame_rpc_system::{System, SystemApiServer};
+
+type BlockNumber = u32;
+type AccountId = AccountId32;
```

Comment:
So these are hardcoded assumptions? I am actually not sure. This will probably work with a runtime that also uses `Multi*` account id types. 

In any case, Let's move all of the types that used to come from the hardcoded runtime, and no longer are, to a `assumptions.rs` (or similar name) and document exactly what assumptions exist, and why.